### ISSUE TITLE
drag/drop image multiple times

### ISSUE THREAD

**Issue by bwmanske** [laurent22/joplin#8055]
My "Notes Attachments" under the tool menu is littered with the same file over and over for the .PNG files I am using.

I consider this a bug as it wastes disk space and time syncing.

I left a note on the forums but it has been ignored for the last 16 days.

## Environment
Windows Intel
Joplin 2.9.17 (prod, win32)

Client ID: e33a656317c54dcea10be27a451197c3
Sync Version: 3
Profile Version: 41
Keychain Supported: Yes

Revision: a84a8e7

## Steps to reproduce

1. Drag image1.png to markup window
2. Drag image1.png to markup window
3. Open the Tool menu and select "Note Attachments"
4. observe 2 identical entries differing by only the UID

## Describe what you expected to happen

I understand why you may wish to create a second entry if the file has been modified. With the same exact file I would expect you to search for the file name within the note attachments and if NOT found then create a new entry under that name and generate a UID.
If the name IS found then compare the contents, if the contents differ then create a new UID and save the data.  If the contents are the SAME then then use the UID of the existing entry and put that into the markdown code.  

<code>
    exitLoop = false
    createEntry = true
    newEntry = new ( )
    newEntry.name = <dropped item name>
    newEntry.data = <droppped item data>
    newEntry.UID = <Generate a new Unique ID>
    noteIndex = 0
    do
        get (currentEntry(noteIndex) )
        if (newEntry.name == currentEntry.name)
            if (newEntry.data == currentEntry.data)
                exitLoop = true
                createEntry = false
                newEntry.UID = currentEntry.UID
            else <nothing>
        else <nothing>
        if (not exitLoop)
            if (noteIndex + 1) >=  <# attachments>
                exitLoop = true
            else
                noteIndex = noteIndex + 1
    while not exitLoop
    if createEntry 
        <proceede normally to create the new attachment>
    else
        <skip creation of the new attachment, insert the markdown>
<code/>

The pseudo code above is to just give you a feeling for this idea. I'm sure it isn't as simple as I think.

Thank you,

**Comment by JackGruber** [comment]
I would say this is not a bug, this is by design. For example, I want to have a new resource created every time I add the image to a note. Since I can also edit each image in the notes independently and if I want the same resource, I just copy the markdown resource link from another note.

**Comment by bwmanske** [comment]
I believe that my suggestion would help out the people in your scenario.  

First let me clear a couple of things up. It wasn't until I saw your reply that I realized that the pseudo code I had written had text deleted from it. Take for example the 2 blank code lines at the end.  Those lines contained notes about what I thought was logical at those points written in plain text. I thing I entered my notes with less than and greater than symbols.  The 2 else statements together had notes on them. The following if statement is not part of the above 2 else statements.

As it stands now you would drag the image to the document 3 times. You would get (Image1.png UID_1) (Image1.png UID_2) (Image1.png UID_3) To modify one of these you are doomed to note the hex number at the location of interest then find the name in note attachments then look down the list of UIDs to find the one you want. The line with the UID is not highlighted with the cursor over it and the table doesn't have alternating rows of background shading. This means that you must find the name on the left side follow it to the right side to look for the correct UID and then follow that line back to the left (all the while trying to stay on the line). 

My suggestion was to change the file name then add it to the attachments.  You would get a second copy of the data but you would have a unique name. This would relieve the dependence on looking at the UID. This way you'd only get a new entry if you intentionally acted to change the name.   The files in the notes attachments could be visually identified by name rather than UID or sequentially opening each file until you find the right one.  

The first line of my post I did say that I understood the reason for multiple copies of the data, I just think this would be better for both use cases.  Anything to lessen the dependence on UIDs is a step forward in my book.

I just tried the less than and greater than symbols to bookend some text and then hit the Preview tab.  Sure enough the text was gone.  If you are interested I will try to recreate my thoughts.  I'm sure that you will choose best for the majority of your users.  Thanks.
